UnitystationLauncher/Services/AuthService.cs

```diff
@@ -0,0 +1,169 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Net.Mail;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using Serilog;
+using UnitystationLauncher.Constants;
+using UnitystationLauncher.Models;
+using UnitystationLauncher.Models.Api;
+using UnitystationLauncher.Models.ConfigFile;
+using UnitystationLauncher.Services.Interface;
+
+namespace UnitystationLauncher.Services
+{
+    public class AuthService
+    {
+        private readonly HttpClient _http;
+        public LoginMsg? LoginMsg { get; set; }
+        public bool AttemptingAutoLogin { get; set; }
+        private readonly IAuthProvider _IAuthProvider;
+        private readonly IPreferencesService _preferencesService;
+
+        public AuthService(HttpClient http, IAuthProvider IAuthProvider, IPreferencesService preferencesService)
+        {
+            _http = http;
+            _IAuthProvider = IAuthProvider;
+            _preferencesService = preferencesService;
+            LoadAuthSettings();
+        }
+
+
+        private string AuthSettingsPath => Path.Combine(_preferencesService.GetPreferences().InstallationPath, "authSettings.json");
+
+        public AccountLoginResponse? AccountLoginResponse { get; set; }
+
+
+        private void LoadAuthSettings()
+        {
+            try
+            {
+                if (File.Exists(AuthSettingsPath))
+                {
+                    var json = File.ReadAllText(AuthSettingsPath);
+                    var AccountLoginResponseA = JsonSerializer.Deserialize<AccountLoginResponse>(json);
+                    AccountLoginResponse = AccountLoginResponseA;
+                }
+            }
+            catch (Exception)
+            {
+                // Something went wrong reading the auth settings. Just ask the user to log in again.
```

Comment:
log the exception or do something with it

Reply:
idk What to do here since this is from the original code and I don't know what happen if I added exception

Reply:
just log it then